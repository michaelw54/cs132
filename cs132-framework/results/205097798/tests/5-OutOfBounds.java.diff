1,4c1,3
< ParseException: Encountered " <IDENTIFIER> "w_2 "" at line 2, column 1.
< Was expecting:
<     "return" ...
<     
---
> 0
> --- Error (Explicit) ---
> array index out of bounds
