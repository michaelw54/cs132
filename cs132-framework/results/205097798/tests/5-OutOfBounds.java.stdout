func Main()
    w4 = 4
    w2 = alloc(w4)
    w4 = 4
    w3 = alloc(w4)
    [w2 + 0] = w3
    w4 = @A_run
    [w3 + 0] = w4
    w1 = @A_run
    w0 = call w1(w2)
    print(w0)
    ret_ = 0
    return ret_
    
func A_run(this)
    a = 0
    w11 = 20
    w9 = 4
    w10 = 1
    w7 = w11 + w10
    w8 = w7 * w9
    w6 = alloc(w8)
    [w6 + 0] = w11
    a = w6
    w17 = a
    w18 = 10
    w14 = [w17 + 0]
    w15 = w18 < w14
    if0 w15 goto outOfBounds_13
    w14 = 4
    w15 = 1
    w16 = w18 + w15
    w15 = w16 * w14
    w14 = w17 + w15
    w13 = [w14 + 0]
    goto pass_13
  outOfBounds_13:
    error("array index out of bounds")
  pass_13:
    print(w13)
    w24 = a
    w25 = 40
    w21 = [w24 + 0]
    w22 = w25 < w21
    if0 w22 goto outOfBounds_20
    w21 = 4
    w22 = 1
    w23 = w25 + w22
    w22 = w23 * w21
    w21 = w24 + w22
    w20 = [w21 + 0]
    goto pass_20
  outOfBounds_20:
    error("array index out of bounds")
  pass_20:
    return w20
    
